tools/Customization/DevHome.Customization/Views/OptimizeDevDriveDialog.xaml.cs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.Input;
+using DevHome.Common.Extensions;
+using DevHome.Customization.CustomControls;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Windows.Storage;
+using Windows.Storage.Pickers;
+using WinUIEx;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace DevHome.Customization.Views;
+
+public sealed partial class OptimizeDevDriveDialog : ContentDialog
+{
+    public OptimizeDevDriveDialog(string cacheToBeMoved, string existingCacheLocation, string environmentVariableToBeSet)
+    {
+        this.InitializeComponent();
+        CacheToBeMoved = cacheToBeMoved;
+        ExistingCacheLocation = existingCacheLocation;
+        EnvironmentVariableToBeSet = environmentVariableToBeSet;
+    }
+
+    public string CacheToBeMoved
+    {
+        get => (string)GetValue(CacheToBeMovedProperty);
+        set => SetValue(CacheToBeMovedProperty, value);
+    }
+
+    public string ExistingCacheLocation
+    {
+        get => (string)GetValue(ExistingCacheLocationProperty);
+        set => SetValue(ExistingCacheLocationProperty, value);
+    }
+
+    public string EnvironmentVariableToBeSet
+    {
+        get => (string)GetValue(EnvironmentVariableToBeSetProperty);
+        set => SetValue(EnvironmentVariableToBeSetProperty, value);
+    }
+
+    private async void OnBrowseButtonClick(object sender, RoutedEventArgs e)
+    {
+        // Create a folder picker dialog
+        var folderPicker = new FolderPicker
+        {
+            SuggestedStartLocation = PickerLocationId.Desktop,
+            ViewMode = PickerViewMode.List,
+        };
+
+        folderPicker.FileTypeFilter.Add("*");
+        WinRT.Interop.InitializeWithWindow.Initialize(folderPicker, Microsoft.UI.Xaml.Application.Current.GetService<WindowEx>().GetWindowHandle());
+        var folder = await folderPicker.PickSingleFolderAsync();
+
+        if (folder != null)
+        {
+            DirectoryPathTextBox.Text = folder.Path;
+        }
+    }
+
+    private void MoveDirectory(string sourceDirectory, string targetDirectory)
+    {
+        try
+        {
+            // Create the target directory if it doesn't exist
+            if (!Directory.Exists(targetDirectory))
+            {
+                Directory.CreateDirectory(targetDirectory);
+            }
+
+            // Get all files and subdirectories in the source directory
+            var files = Directory.GetFiles(sourceDirectory);
+            var subdirectories = Directory.GetDirectories(sourceDirectory);
+
+            // Move files
+            foreach (var file in files)

```

Comment:
Recursion?  does [Directory.Move](https://learn.microsoft.com/dotnet/api/system.io.directory.move?view=net-8.0) not work?

Reply:
Directory.Move does not work across volumes.